File: spring-boot-backend/src/main/java/com/springbootbackend/model/Order.java
package com.springbootbackend.model;

import java.util.List;

import jakarta.persistence.*;
import lombok.*;

@Data
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "orders")
public class Order {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long orderId;

    @ManyToOne
    @JoinColumn(name = "user_id",nullable = false)
    private UserEntity userEntity;

    @OneToMany(mappedBy ="order",cascade = CascadeType.ALL,orphanRemoval = true)
    private List<OrderItem> orderItems;

    private Double totalPrice;

    @Enumerated(EnumType.STRING)
    private OrderStatus status = OrderStatus.PENDING;
}

File: flask-backend/models/order.py
from extensions import db
import enum
from sqlalchemy import Enum


class OrderStatus(enum.Enum):
    PENDING = "PENDING"
    DELIVERED = "DELIVERED"
    CANCELLED = "CANCELLED"
    PROCESSING = "PROCESSING"
    SHIPPED = "SHIPPED"


class Order(db.Model):

    __tablename__ = "orders"

    orderId = db.Column(db.Integer, primary_key=True)
    userId = db.Column(db.Integer, db.ForeignKey("users.userId"), nullable=False)
    total_price = db.Column(db.Float, default=0.0)
    status = db.Column(Enum(OrderStatus), default=OrderStatus.PENDING)

    # Relationship to OrderItem
    items = db.relationship("OrderItem", backref="order", cascade="all, delete-orphan")
